src/smallestPrimeFactor.js

```diff
@@ -12,11 +12,30 @@
  * @returns {number} The smallest prime factor of num
  */
 function smallestPrimeFactor(num) {
-  // This is your job. :)
-}
+let arr = [];
+
+for ( var i = 2; i < num; i++) {
+let isPrime;
 
+if (num % i === 0) {
```

Comment:
The first value of `i` for which this is `true` will be the smallest divisor of `num`. This code tells me you know it must be prime, too.

But that means the first `i` for which this is `true` _is_ the smallest prime factor! Just return `i` here.

If the loop ends without returning early, return `num` itself.

That is, the smallest number greater than `1` that divides `num` _must_ be prime. Call that number `k`. If `k` weren't prime then it would have a factor greater than `1` and smaller than `k`. But that factor would also have divided `num` and we would've found it earlier.

The smallest factor of `num` _must_ be prime (so finding the smallest _prime_ factor is no different than finding the smallest factor in general).